File: src/main/java/org/dukcode/ps/codetree/trail02/chapter05/lesson01/IntroContinuousNumber2.java
package org.dukcode.ps.codetree.trail02.chapter05.lesson01;

import java.io.BufferedReader;
import java.io.BufferedWriter;
import java.io.IOException;
import java.io.InputStreamReader;
import java.io.OutputStreamWriter;

/**
 * @see <a
 * href="https://www.codetree.ai/ko/trails/complete/curated-cards/intro-continuous-number2/description">LINK</a>
 */
public class IntroContinuousNumber2 {

  private static BufferedReader br;
  private static BufferedWriter bw;

  public static void main(String[] args) throws IOException {
    br = new BufferedReader(new InputStreamReader(System.in));
    bw = new BufferedWriter(new OutputStreamWriter(System.out));

    int n = Integer.parseInt(br.readLine());

    int maxCnt = 0;
    int before = -1;
    int cnt = 0;
    for (int i = 0; i < n; i++) {
      int num = Integer.parseInt(br.readLine());

      if (before != num) {
        maxCnt = Math.max(maxCnt, cnt);
        before = num;
        cnt = 1;
        continue;
      }

      cnt++;
    }

    maxCnt = Math.max(maxCnt, cnt);

    bw.write(String.valueOf(maxCnt));

    br.close();
    bw.close();

  }
}

File: src/main/kotlin/org/dukcode/ps/codetree/trail02/chapter05/lesson01/IntroContinuousNumber2.kt
package org.dukcode.ps.codetree.trail02.chapter05.lesson01.introContinuousNumber2

import java.io.BufferedReader
import java.io.BufferedWriter
import java.io.InputStreamReader
import java.io.OutputStreamWriter

private val br = BufferedReader(InputStreamReader(System.`in`))
private val bw = BufferedWriter(OutputStreamWriter(System.out))

/**
 * @see <a href="https://www.codetree.ai/ko/trails/complete/curated-cards/intro-continuous-number2/description">LINK</a>
 */
fun main() {
    val n = br.readLine().toInt()

    val arr =
        IntArray(n) {
            br.readLine().toInt()
        }

    var maxCnt = 0
    var cnt = 0
    for (idx in arr.indices) {
        if (idx == 0 || arr[idx - 1] != arr[idx]) {
            maxCnt = maxOf(maxCnt, cnt)
            cnt = 1
            continue
        }

        if (arr[idx - 1] == arr[idx]) {
            cnt++
        }
    }

    maxCnt = maxOf(maxCnt, cnt)

    bw.write(maxCnt.toString())

    bw.flush()
}
